pages/checkin.js

```diff
@@ -118,17 +118,23 @@ class Checkin extends React.Component {
       .then((lastVisitedTimes) => {
         this.setState({lastScannedID:e.target.calID.value, visitsLastWeek:lastVisitedTimes, lastScannedTime:new Date().toLocaleTimeString()})
         this.showSuccess("Sucessfully scanned ID: " + e.target.calID.value,1000)
-        e.target.calID.value = null;
-        document.getElementById("calID").focus();
+        fieldsetCalId.disabled = false
+        e.target.calID.value = null
+        calIdTextBox.focus();
       })
       .catch((err) => {
+        fieldsetCalId.disabled = false
+        e.target.calID.value = null
+        calIdTextBox.focus()
         this.showError("Failed scanning ID: " + e.target.calID.value,3000)
       });
     })
     .catch((err) => {
+      fieldsetCalId.disabled = false
+      e.target.calID.value = null;
+      calIdTextBox.focus();
       this.showError("Failed scanning ID: " + e.target.calID.value,300)
-    })
-  
+    })  
```

Comment:
wait actually i think a `.finally()` after this line would work best

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally